### ISSUE TITLE
[java] InvalidLogMessageFormat: False positive with lombok @Value generated methods

### ISSUE THREAD

**Issue by altaiezior** [pmd/pmd#5787]
**Affects PMD Version:** 7.14.0

**Rule:** [InvalidLogMessageFormat](https://pmd.github.io/pmd/pmd_rules_java_errorprone.html#invalidlogmessageformat)

**Description:**
If there is no argument but only the log message and the exception then pmd reports `InvalidLogMessageFormat`

**Code Sample demonstrating the issue:**

```java
@Slf4j
public class Main {
    public static void main(String[] args) {
        var abcd = new AClass(new Exception("abcd"));
        log.error("Hello World", abcd.getT());
    }

    @Value
    private static class AClass {
        Throwable t;
    }
}
```

**Expected outcome:**

PMD reports a violation at line ..., but that's wrong. That's a false positive.

**Running PMD through:** *Maven*

**Comment by adangel** [comment]
Hi @anuragagarwal561994 , thanks for reporting.
Which logging framework are you using?
I cannot reproduce it with slf4j or log4j2.

**Comment by altaiezior** [comment]
Let me try and make a reproducer for the same, I think it is another issue.

```
class Wrapper {
    Throwable tx;

    public Throwable getException() {
         return tx;
    }
}
log.error("Hello World", wrapper.getException())
```

But I will try to write a proper reproducer for the same, I think when I am fetching it as a field of a class, pmd is realising it to be a variable instead of a throwable and hence creating this issue.

I am using `slf4j` as api and `log4j2` as impl.

**Comment by altaiezior** [comment]
@adangel 

```
@Slf4j
public class Main {
    public static void main(String[] args) {
        var abcd = new AClass(new Exception("abcd"));
        log.error("Hello World", abcd.getT());
    }

    @Value
    private static class AClass {
        Throwable t;
    }
}
```

This is a very simple reproducer of the above issue, note that when using without class directly new Exception(), the issue is not there, but when the exception is wrapped within a class then there is an issue

**Comment by oowekyala** [comment]
The problem appears to be the use of `@lombok.Value`. You are calling a getter that was declared by Lombok, and pmd doesn't know about it. That could be part of #3119

**Comment by altaiezior** [comment]
@oowekyala but it was not breaking earlier, do you think something was introduced in the release 7.14

**Comment by adangel** [comment]
I guess, since https://github.com/pmd/pmd/issues/5702 has been fixed, PMD now recognizes that `log` is actually a logger (due to better slf4j support). Before that, the logging messages were not analyzed at all, because PMD didn't even know what `log` is.

**Comment by altaiezior** [comment]
@adangel I tried a few variations which work if I explicitly type-case like
```
 log.error("Hello World", (Throwable) abcd.getT());

// or

Throwable t = abcd.getT();
log.error("Hello World", t);
```

The above rule works correctly but even if I do something like

```
var t = abcd.getT();
log.error("Hello World", t)
```

It doesn't work, I guess the `var` variation would actually work for my use-case but I don't know if there are bottlenecks in resolving that as issue.

I am not able to upgrade my `pmd-java` version since `7.13`
